File: app/src/main/java/viewmodel/AddEditCategoryViewModel.java
package viewmodel;

import androidx.annotation.StringRes;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import java.util.Calendar;
import java.util.Objects;

import hlv.cute.todo.R;
import model.Category;
import model.DateTime;
import model.Todo;
import utils.ResourceUtils;

public class AddEditCategoryViewModel extends ViewModel {

    private Category category;
    private boolean isEditMode = false;

    public void setCategory(Category category) {
        this.category = category;
    }

    public Category getCategory() {
        return category;
    }

    public boolean isEditMode() {
        return isEditMode;
    }

    public void setEditMode(boolean editMode) {
        isEditMode = editMode;
    }

    public Category addCategory(String title) {
        category = new Category();
        category.setId(category.getId());
        category.setName(title);
        return category;
    }

    public Category editCategory(String title) {
        Category mustBeEditedCategory = new Category();
        mustBeEditedCategory.setId(category.getId());
        mustBeEditedCategory.setName(title);
        return mustBeEditedCategory;
    }

    //region: get text and strings
    public String getTitleFragment() {
        return getString(isEditMode ? R.string.edit_category : R.string.add_new_category);
    }

    public String getButtonPrimaryText() {
        return getString(isEditMode ? R.string.edit : R.string.save);
    }

    public String getEditTextTitle() {
        return isEditMode ? category.getName() : "";
    }

    //endregion

    private String getString(@StringRes int stringRes) {
        return ResourceUtils.get().getString(stringRes);
    }

}

File: app/src/main/java/viewmodel/AddEditCategoryViewModel.kt
package viewmodel

import androidx.lifecycle.ViewModel
import dagger.assisted.Assisted
import dagger.hilt.android.lifecycle.HiltViewModel
import hlv.cute.todo.App
import hlv.cute.todo.R
import model.Category
import utils.AppResourcesProvider
import javax.inject.Inject

@HiltViewModel
class AddEditCategoryViewModel @Inject constructor(
    private val provideResource: AppResourcesProvider
) : ViewModel() {

    var category: Category? = null
    var isEditMode = false

    fun addCategory(title: String?): Category {
        category = Category(name = title)
        return category!!
    }

    fun editCategory(title: String?): Category {
        val mustBeEditedCategory = Category().apply {
            id = category!!.id
            name = title
        }

        return mustBeEditedCategory
    }

    fun getTitleFragment(): String =
        provideResource.getString(if (isEditMode) R.string.edit_category else R.string.add_new_category)

    fun getButtonPrimaryText(): String =
        provideResource.getString(if (isEditMode) R.string.edit else R.string.save)

    fun getEditTextTitle(): String? = if (isEditMode) category!!.name else ""

}